rllib/agents/trainer_template.py

```diff
@@ -92,8 +93,14 @@ class trainer_cls(base):
         _default_config = default_config or COMMON_CONFIG
         _policy_class = default_policy
 
-        def __init__(self, config=None, env=None, logger_creator=None):
-            Trainer.__init__(self, config, env, logger_creator)
+        def __init__(self,
+                     config: TrainerConfigDict = None,
+                     env: Union[str, EnvType, None] = None,
+                     logger_creator: Callable[[], Logger] = None,
+                     remote_checkpoint_dir: Optional[str] = None,
+                     sync_function_tpl: Optional[str] = None):
+            Trainer.__init__(self, config, env, logger_creator,
+                             remote_checkpoint_dir, sync_function_tpl)
```

Comment:
This is just adding a couple missed API entrypoints right?

Reply:
This is the actual fix that is needed to make RLLib work with cloud checkpointing.

Yes, it is just making sure that the constructor of the `Trainer*Templates*` calls the full constructor of the `Trainer`

Reply:
Looks very good. Thanks for the fix @krfricke !